src/main/java/org/sonar/plugins/stash/issue/collector/StashCollector.java

```diff
@@ -42,53 +42,56 @@ public static StashCommentReport extractComments(JSONObject jsonComments) throws
     return result;
   }
   
-  public static StashComment extractComment(JSONObject jsonComment, String path, Long line) throws StashReportExtractionException {
-    StashComment result = null;
+  public static StashComment extractComment(JSONObject jsonComment, String path, Long line)
+         throws StashReportExtractionException {
 
-    long id = (long) jsonComment.get("id");
-    String message = (String) jsonComment.get("text");
+    long id = (long)jsonComment.get("id");
+    String message = (String)jsonComment.get("text");
 
-    long version = (long) jsonComment.get(VERSION);
+    long version = (long)jsonComment.get(VERSION);
 
-    JSONObject jsonAuthor = (JSONObject) jsonComment.get(AUTHOR);
+    JSONObject jsonAuthor = (JSONObject)jsonComment.get(AUTHOR);
     StashUser stashUser = extractUser(jsonAuthor);
 
-    result = new StashComment(id, message, path, line, stashUser, version);
+    StashComment result = new StashComment(id, message, path, line, stashUser, version);
 
     return result;
   }
   
   public static StashComment extractComment(JSONObject jsonComment) throws StashReportExtractionException {
-    StashComment result = null;
 
-    JSONObject jsonAnchor = (JSONObject) jsonComment.get("anchor");
+    JSONObject jsonAnchor = (JSONObject)jsonComment.get("anchor");
     if (jsonAnchor == null) {
-      throw new StashReportExtractionException("JSON Comment does not contain any \"anchor\" tag to describe comment \"line\" and \"path\"");
+      throw new StashReportExtractionException("JSON Comment does not contain any \"anchor\" tag"
+                                               + " to describe comment \"line\" and \"path\"");
     }
 
-    String path = (String) jsonAnchor.get("path");
+    String path = (String)jsonAnchor.get("path");
 
     // can be null if comment is attached to the global file
-    Long line = (Long) jsonAnchor.get("line");
+    Long line = (Long)jsonAnchor.get("line");
 
-    result = extractComment(jsonComment, path, line);
+    StashComment result = extractComment(jsonComment, path, line);

```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1488)